hitmap.go

```diff
@@ -49,6 +50,7 @@ func (hits *hitMap) expireLimits() {
 	for {
 		hitMapCopy := hits.getMap()
 		for _, ipr := range hitMapCopy {
+			// TODO address concurrent read
```

Comment:
Yeah, ipr does have a mutex. I did put a lock here but it is awkward because I need to unlock before calling `RateLimit()`. Thus the `TODO`.